### DISCUSSION TITLE
Adding role property to user.session - Profile id is missing in Google OAuth profile response

### DISCUSSION THREAD

**Discussion by fatalcaster** [nextauthjs/next-auth#7547] [Help] [answered]
'm following this [tutorial](https://authjs.dev/guides/basics/role-based-access-control) on how to add roles in next-auth session. Unfortunately, when I add profile property, I get undefined behavior of the profile missing. There are also errors regarding typescript. Is this an error on my side, or is it a known bug, since I couldn't find anything on it.

Here's my code so far:

```tsx
export const authOptions: AuthOptions = {
  secret: process.env.NEXT_PUBLIC_SECRET!,
  providers: [
    GoogleProvider({
      clientId: process.env.GOOGLE_CLIENT_ID!,
      clientSecret: process.env.GOOGLE_CLIENT_SECRET!,
      // profile: async (profile) => {
      //   return { ...profile, role: profile.role ?? Role.USER };
      // },
    }),
  ],
  pages: {
    signIn: "/",
  },

  adapter: PrismaAdapter(prisma),
};

```
as you can see it's the same as the one from the tutorial, when I comment out the profile section I get the expected behavior without role. Any help would be appreciated!

Version of Next.js: 13.4.1 (app directory)

**Comment by 418e** [comment, marked as answer]
I have had the same problem and solved it by adding the "or" operator.

```js
      clientSecret: process.env.GITHUB_SECRET,
      profile(profile, tokens) {
        return {
          // before:    
          // id: tokens.id_token
          // after:
          id: tokens.id_token || profile.id,
         // ...
        };
      },
```
I hope it will help.
